habitat-baselines/habitat_baselines/rl/multi_agent/multi_agent_access_mgr.py

```diff
@@ -273,4 +282,31 @@ def updater(self):
 
     @property
     def policy_action_space(self):
-        return self.actor_critic.policy_action_space
+        # TODO: Hack for discrete HL action spaces.
+        return spaces.MultiDiscrete(
+            tuple([agent.policy_action_space.n for agent in self._agents])
+        )
+
+    def update_hidden_state(self, rnn_hxs, prev_actions, action_data):
+        """
+        Update the hidden state of the agents in the population. Writes to the
+        data in place.
+        """
+        n_agents = len(self._active_agents)
+        hxs_dim = rnn_hxs.shape[-1] // n_agents
+        ac_dim = prev_actions.shape[-1] // n_agents
+        # Not very efficient, but update each agent's hidden state individually.
+        for env_i, should_insert in enumerate(action_data.should_inserts):
+            for policy_i, agent_should_insert in enumerate(should_insert):
+                if agent_should_insert.item():
+                    rnn_sel = slice(
+                        policy_i * hxs_dim, (policy_i + 1) * hxs_dim
+                    )
+                    rnn_hxs[env_i, :, rnn_sel] = action_data.rnn_hidden_states[
+                        env_i, :, rnn_sel
+                    ]
+
+                    ac_sel = slice(policy_i * ac_dim, (policy_i + 1) * ac_dim)
+                    prev_actions[env_i, ac_sel].copy_(
+                        action_data.actions[env_i, ac_sel]
+                    )
```

Comment:
```suggestion
                if not agent_should_insert.item():
                    continue
                rnn_sel = slice(
                    policy_i * hxs_dim, (policy_i + 1) * hxs_dim
                )
                rnn_hxs[env_i, :, rnn_sel] = action_data.rnn_hidden_states[
                    env_i, :, rnn_sel
                ]

                ac_sel = slice(policy_i * ac_dim, (policy_i + 1) * ac_dim)
                prev_actions[env_i, ac_sel].copy_(
                    action_data.actions[env_i, ac_sel]
                )
```
The code is already very nested. Consider using continue (similar to the code above) to reduce nesting.